### PR TITLE
PAPP-35152: changing documentation to config variables section to be …

### PR DESCRIPTION
**Pull request by tapishj-splunk**
- Changing the Configuration Variables description after doc reviews for Talos
- Changing so hidden config params don't show up on READme docs

### CODE REVIEW

File: github-actions/start-release/templates/connector_detail.md [hunk 1]
```diff
@@ -19,12 +19,12 @@ Minimum Product Version: {{connector.min_phantom_version}}
 {%- if connector.configuration %}
 
 ### Configuration Variables
-The below configuration variables are required for this Connector to operate.  These variables are specified when configuring a {{connector.product_name}} asset in SOAR.
+This table lists the configuration variables required to operare {{connector.name}}. These variables are specified when configuring a {{connector.product_name}} asset in SOAR.
```
Comment (phantom-jacob) [inline github-actions/start-release/templates/connector_detail.md:22]:
```suggestion
This table lists the configuration variables required to operate {{connector.name}}. These variables are specified when configuring a {{connector.product_name}} asset in Splunk SOAR.
```

File: github-actions/start-release/templates/connector_detail.md [hunk 2]
```diff
@@ -19,12 +19,12 @@ Minimum Product Version: {{connector.min_phantom_version}}
...
 
 VARIABLE | REQUIRED | TYPE | DESCRIPTION
 -------- | -------- | ---- | -----------
 {%- for name, parameters in connector.configuration.items() %}
-{%- if name != "ph" %}
+{%- if name != "ph" and ("visibility" not in parameters or parameters.visibility|length > 0) %}
```
Comment (phantom-jacob) [inline github-actions/start-release/templates/connector_detail.md:27]: Would this have prevented this issue? https://github.com/splunk-soar-connectors/msgraphforoffice365/pull/44/files#r1854866249

File: github-actions/start-release/tests/data/build_docs/has_existing_readme/expected_readme.md [hunk 3]
```diff
@@ -613,7 +613,7 @@ explanation in Overview, and some individuals Apps have their sections.
 
 
 ### Configuration Variables
-The below configuration variables are required for this Connector to operate.  These variables are specified when configuring a QRadar asset in SOAR.
+This table lists the configuration variables required to operare QRadar. These variables are specified when configuring a QRadar asset in SOAR.
```
Comment (phantom-jacob) [inline github-actions/start-release/tests/data/build_docs/has_existing_readme/expected_readme.md:616]:
```suggestion
This table lists the configuration variables required to operate QRadar. These variables are specified when configuring a QRadar asset in Splunk SOAR.
```

File: github-actions/start-release/tests/data/build_docs/has_existing_readme_html_and_autogen_md/expected_readme.md [hunk 4]
```diff
@@ -613,7 +613,7 @@ explanation in Overview, and some individuals Apps have their sections.
 
 
 ### Configuration Variables
-The below configuration variables are required for this Connector to operate.  These variables are specified when configuring a QRadar asset in SOAR.
+This table lists the configuration variables required to operare QRadar. These variables are specified when configuring a QRadar asset in SOAR.
```
Comment (phantom-jacob) [inline github-actions/start-release/tests/data/build_docs/has_existing_readme_html_and_autogen_md/expected_readme.md:616]:
```suggestion
This table lists the configuration variables required to operate QRadar. These variables are specified when configuring a QRadar asset in Splunk SOAR.
```

File: github-actions/start-release/tests/data/build_docs/has_existing_readme_html_and_md/expected_readme.md [hunk 5]
```diff
@@ -613,7 +613,7 @@ explanation in Overview, and some individuals Apps have their sections.
 
 
 ### Configuration Variables
-The below configuration variables are required for this Connector to operate.  These variables are specified when configuring a QRadar asset in SOAR.
+This table lists the configuration variables required to operare QRadar. These variables are specified when configuring a QRadar asset in SOAR.
```
Comment (phantom-jacob) [inline github-actions/start-release/tests/data/build_docs/has_existing_readme_html_and_md/expected_readme.md:616]:
```suggestion
This table lists the configuration variables required to operate QRadar. These variables are specified when configuring a QRadar asset in Splunk SOAR.
```

File: github-actions/start-release/tests/data/build_docs/has_no_readme/expected_readme.md [hunk 6]
```diff
@@ -11,7 +11,7 @@ Minimum Product Version: 4.9.39220
 This app supports generic, investigative, and ingestion actions on an IBM QRadar device
 
 ### Configuration Variables
-The below configuration variables are required for this Connector to operate.  These variables are specified when configuring a QRadar asset in SOAR.
+This table lists the configuration variables required to operare QRadar. These variables are specified when configuring a QRadar asset in SOAR.
```
Comment (phantom-jacob) [inline github-actions/start-release/tests/data/build_docs/has_no_readme/expected_readme.md:14]:
```suggestion
This table lists the configuration variables required to operate QRadar. These variables are specified when configuring a QRadar asset in Splunk SOAR.
```

File: github-actions/start-release/tests/data/build_docs/has_no_readme_set_app_version/expected_readme.md [hunk 7]
```diff
@@ -11,7 +11,7 @@ Minimum Product Version: 4.9.39220
 This app supports generic, investigative, and ingestion actions on an IBM QRadar device
 
 ### Configuration Variables
-The below configuration variables are required for this Connector to operate.  These variables are specified when configuring a QRadar asset in SOAR.
+This table lists the configuration variables required to operare QRadar. These variables are specified when configuring a QRadar asset in SOAR.
```
Comment (phantom-jacob) [inline github-actions/start-release/tests/data/build_docs/has_no_readme_set_app_version/expected_readme.md:14]:
```suggestion
This table lists the configuration variables required to operate QRadar. These variables are specified when configuring a QRadar asset in Splunk SOAR.
```

File: github-actions/start-release/tests/data/readme_to_markdown/has_no_readme/expected_readme.md [hunk 8]
```diff
@@ -11,7 +11,7 @@ Minimum Product Version: 4\.9\.39220
 This app supports generic, investigative, and ingestion actions on an IBM QRadar device
 
 ### Configuration Variables
-The below configuration variables are required for this Connector to operate.  These variables are specified when configuring a QRadar asset in SOAR.
+This table lists the configuration variables required to operare QRadar. These variables are specified when configuring a QRadar asset in SOAR.
```
Comment (phantom-jacob) [inline github-actions/start-release/tests/data/readme_to_markdown/has_no_readme/expected_readme.md:14]:
```suggestion
This table lists the configuration variables required to operate QRadar. These variables are specified when configuring a QRadar asset in Splunk SOAR.
```

**Review by phantom-jacob — CHANGES_REQUESTED** [review]

File: github-actions/start-release/templates/connector_detail.md (hunk 2 above)
Comment (tapishj-splunk) [reply github-actions/start-release/templates/connector_detail.md:27]: it wouldn't but the condition should change to include parameters.data_type != "ph" as long as a ph data_type should be hidden.

Comment (phantom-jacob) [reply github-actions/start-release/templates/connector_detail.md]: Then let's change the condition? The `ph` data type exists solely to make UI forms look better. It's a nasty hack

**Review by phantom-jacob — APPROVED** [review]

**Pull request merged**